doc/dnf_daemon/examples/rpm_list_fd.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python3
+# Copyright Contributors to the libdnf project.
+#
+# This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+#
+# Libdnf is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Libdnf is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+This is an example how to use `list_fd()` method of dnf5daemon-server Rpm interface.
+"""
+
+import dbus
+import json
+import os
+import select
+import time
+
+DNFDAEMON_BUS_NAME = 'org.rpm.dnf.v0'
+DNFDAEMON_OBJECT_PATH = '/' + DNFDAEMON_BUS_NAME.replace('.', '/')
+
+IFACE_SESSION_MANAGER = '{}.SessionManager'.format(DNFDAEMON_BUS_NAME)
+IFACE_RPM = '{}.rpm.Rpm'.format(DNFDAEMON_BUS_NAME)
+
+
+def repoquery(iface_rpm, options):
+    """Generator function that yields packages as they arrive from the server."""
+
+    # create a pipe and pass the write end to the server
+    pipe_r, pipe_w = os.pipe()
+    # transfer id serves as an identifier of the pipe transfer for a signal emitted
+    # after server finish. This example to not use it.
+    iface_rpm.list_fd(options, pipe_w, "the_transfer_id")
+    # close the write end - otherwise poll cannot detect the end of transmission
+    os.close(pipe_w)
```

Comment:
What happen when DBus is overloaded and pipe_w is closed before server part receive DBus message? Or file descriptor is already cloned for server?

Reply:
The `list_fd()` call here is synchronous - it will wait for the reply from D-Bus.